charts/castai-hosted-model/child-charts/vllm/templates/deployment.yaml

```diff
@@ -0,0 +1,78 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "vllm.fullname" . }}
+  labels:
+    {{- include "vllm.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:                                                                                                                                  
+    matchLabels:
+      {{- include "vllm.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "vllm.labels" . | nindent 8 }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      containers:
+        - name: "vllm"
+          image: "{{ required "Required value 'image.repository' must be defined" .Values.image.repository }}:{{ required "Required value 'image.tag' must be defined" .Values.image.tag }}"
+          imagePullPolicy: IfNotPresent
+          command: ["/bin/sh", "-c", "vllm", "serve"]
+          args:
+            - {{ .Values.model | quote }}
+            - "--trust-remote-code"
+            - "--gpu-memory-utilization=1"
+            - "--task=generate"
+            - "--dtype=half"
```

Comment:
Mostly just curious, why this and not leave it "auto"?

Reply:
It appears that when left with `auto`, the model may fail to run as only specific types of GPUs would be usable for it. For example, I tried running a llama3.2:3b model with `dtype=auto` and `nvidia-tesla-t4` GPU, and it would fail on startup because that GPU didn't support that specific configuration. Changing it to `half` was due to the suggestion in the vLLM logs themselves.